src/analyze/gui/parser.cpp

```diff
@@ -653,7 +658,37 @@ void Parser::parseImpl(const QString& path, const QString& diffBase, const Filte
 
         const auto isReparsing = data == oldData;
 
-        emit progressMessageAvailable(isReparsing ? i18n("reparsing data...") : i18n("parsing data..."));
+        auto parsingMsg = isReparsing ? i18n("reparsing data...") : i18n("parsing data...");
+        emit progressMessageAvailable(parsingMsg);
+
+        const auto numPasses = data->stringCache.diffMode ? 2 : 3;
+        auto updateProgress = [this, numPasses, parsingMsg](std::shared_ptr<const ParserData> const & data, time_point start){
+            auto passCompletion = 1.0*data->parsingState.compressedByte/data->parsingState.fileSize;
+            auto totalCompletion = ((data->parsingState.pass + passCompletion)/numPasses);
+            auto spentTime = steady_clock::now() - start;
+            auto spentTime_s = duration_cast<duration<double>>(spentTime).count();
+            auto totalCompletionPerSec = totalCompletion / spentTime_s;
+            auto passCompletionPerSec = passCompletion / (spentTime_s - (data->parsingState.pass/totalCompletionPerSec)) ;
+            auto passRemainingTime_s = (1.0 - passCompletion) / passCompletionPerSec;
+            auto totalRemainingTime_s = (spentTime_s / totalCompletion) * (1.0 - totalCompletion);
+            auto message = QString(
+                parsingMsg
+                    + i18n("\nseconds remaing: ") + QString::number(static_cast<int>(totalRemainingTime_s))
```

Comment:
here and below: please don't do i18n string puzzles, that's hard to translate. instead, do something like this: `i18n("time remaining: %1", Util::formatTime(totalRemainingTime))`

also note the typo in "remaining" in the text

Reply:
This was a result of not quite understanding how `i18n` was used, and slowly expanding what was there. 

Will change.